convert/gitlab/yaml/pipeline.go

```diff
@@ -40,3 +48,188 @@ type (
 		Timeout       string        `yaml:"duration,omitempty"`
 	}
 )
+
+var globalKeys = map[string]struct{}{
+	"after_script":  {},
+	"artifacts":     {},
+	"before_script": {},
+	"cache":         {},
+	"image":         {},
+	"interruptible": {},
+	"retry":         {},
+	"services":      {},
+	"tags":          {},
+	"timeout":       {},
+	"variables":     {},
+	"workflow":      {},
+	"stages":        {},
+	"include":       {},
+}
+
+func (p *Pipeline) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
I think we should be able to handle this parsing without having to track global keys / have a switch statement int he parser.  I will provide suggestions when I have more time to dig in

Reply:
See https://github.com/drone/go-convert/pull/143#issuecomment-1642692837